src/api/photos/usecase.ts

```diff
@@ -157,9 +155,19 @@ export class PhotosUsecase {
   }
 
   private async changePhotoStatus(photoId: PhotoId, newStatus: PhotoStatus): Promise<void> {
-    await this.photosRepository.updateById(photoId, { 
-      status: newStatus,
-      statusChangedAt: new Date()
-    });
+    const photo = await this.photosRepository.getById(photoId);
+    if (photo) {
+      if (newStatus === PhotoStatus.Deleted) {
```

Comment:
same here, a transaction could be a better approach, just to ensure the integrity of this concrete use case